src/Framework/AspNetCoreAnalyzers/src/Analyzers/RouteEmbeddedLanguage/Infrastructure/RouteStringSyntaxDetector.cs

```diff
@@ -0,0 +1,431 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.AspNetCore.Analyzers.RouteEmbeddedLanguage.Infrastructure;
+
+internal static class RouteStringSyntaxDetector
```

Comment:
hrmm... why do you need this?  one of hte poitns of hte current roslyn approach is that *we* do this work for you, and then we call directly into your features to work on the literals we've already found that are properly being passed to things with teh "Route" syntax attribute.

Reply:
RouteStringSyntaxDetector is used by the analyzer and completion provider.

Reply:
even if we still need that, i'd far prefer that we just expose the detector as something in our EA layer so you can just leverage our code, not reimplement it (and have to keep up with fixes in it).

Reply:
This can go away once https://github.com/dotnet/roslyn/pull/62502 goes in.

Reply:
Need to keep it to avoid dependency on external access assembly in analyzer.

Reply:
damn.  that really sucks.  @jaredpar the inability to reference stuff sanely leads to just an exceptional amount of code duplication.  Is there some way we can make this better.  in roslyn this is like a dozen lines.  copying out here explodes it to 400+ lines.  and it means any bug fixes we make have to be done here as well.